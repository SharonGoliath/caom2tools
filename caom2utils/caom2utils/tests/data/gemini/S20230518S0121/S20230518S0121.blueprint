Observation.observationID = ['OBSID'], default = None
Observation.type = OBJECT
Observation.intent = science
Observation.metaRelease = 2023-05-18 09:12:54
Observation.metaProducer = GEMINI2caom2/0.0.0
Observation.algorithm.name = exposure
Observation.instrument.name = GHOST
Observation.telescope.geoLocationX = 1820184.055747326
Observation.telescope.geoLocationY = -5208320.066618682
Observation.telescope.geoLocationZ = -3194829.197152751
Observation.environment.ambientTemp = ['TEMPERAT'], default = None
Plane.productID = S20230518S0121
Plane.metaRelease = 2023-05-18 09:12:54
Plane.dataRelease = 2023-05-18 09:12:54
Plane.dataProductType = image
Plane.calibrationLevel = 1
Plane.metaProducer = GEMINI2caom2/0.0.0
Plane.provenance.name = Gemini Observatory Data
Plane.provenance.project = Gemini Archive
Plane.provenance.producer = ['IMAGESWV', 'ORIGIN'], default = Gemini Observatory
Plane.provenance.reference = http://archive.gemini.edu/searchform/GS-2023A-SV-101-13-009
Plane.provenance.lastExecuted = ['DATE-FTS'], default = None
Artifact.productType = science
Artifact.releaseType = data
Artifact.uri = gemini:GEMINI/S20230518S0121.fits
Artifact.metaProducer = GEMINI2caom2/0.0.0
Chunk = include
Chunk.metaProducer = GEMINI2caom2/0.0.0
Chunk.energy.specsys = TOPOCENT
Chunk.time.exposure = get_exposure(uri)
extension 1:
Chunk.time.exposure = get_exposure(uri)
Chunk.time.resolution = get_exposure(uri)
Chunk.time.axis.axis.ctype = TIME
Chunk.time.axis.axis.cunit = d
Chunk.time.axis.error.syser = 1e-07
Chunk.time.axis.error.rnder = 1e-07
Chunk.time.axis.function.naxis = 1
Chunk.time.axis.function.delta = get_time_delta(header)
Chunk.time.axis.function.refCoord.pix = 0.5
Chunk.time.axis.function.refCoord.val = get_time_function_val(header)
# extension 2:
# Chunk.energy.specsys = TOPOCENT
# Chunk.energy.axis.axis.ctype = WAVE
# Chunk.energy.axis.axis.cunit = nm
# Chunk.energy.axis.range.start.pix = 0.5
# Chunk.energy.axis.range.start.val = _get_energy_chunk_range_start_val(header)
# Chunk.energy.axis.range.end.pix = 1.5
# Chunk.energy.axis.range.end.val = _get_energy_chunk_range_start_val(header)
extension 3:
Chunk.energy.specsys = TOPOCENT
Chunk.energy.axis.axis.ctype = WAVE
Chunk.energy.axis.axis.cunit = nm
Chunk.energy.axis.range.start.pix = 0.5
Chunk.energy.axis.range.start.val = _get_energy_chunk_range_start_val(header)
Chunk.energy.axis.range.end.pix = 1.5
Chunk.energy.axis.range.end.val = _get_energy_chunk_range_start_val(header)
extension 4:
Chunk.energy.specsys = TOPOCENT
Chunk.energy.axis.axis.ctype = WAVE
Chunk.energy.axis.axis.cunit = nm
Chunk.energy.axis.range.start.pix = 0.5
Chunk.energy.axis.range.start.val = _get_energy_chunk_range_start_val(header)
Chunk.energy.axis.range.end.pix = 1.5
Chunk.energy.axis.range.end.val = _get_energy_chunk_range_start_val(header)
extension 5:
Chunk.energy.specsys = TOPOCENT
Chunk.energy.axis.axis.ctype = WAVE
Chunk.energy.axis.axis.cunit = nm
Chunk.energy.axis.range.start.pix = 0.5
Chunk.energy.axis.range.start.val = _get_energy_chunk_range_start_val(header)
Chunk.energy.axis.range.end.pix = 1.5
Chunk.energy.axis.range.end.val = _get_energy_chunk_range_start_val(header)
extension 6:
Chunk.energy.specsys = TOPOCENT
Chunk.energy.axis.axis.ctype = WAVE
Chunk.energy.axis.axis.cunit = nm
Chunk.energy.axis.range.start.pix = 0.5
Chunk.energy.axis.range.start.val = _get_energy_chunk_range_start_val(header)
Chunk.energy.axis.range.end.pix = 1.5
Chunk.energy.axis.range.end.val = _get_energy_chunk_range_start_val(header)
extension 8:
Chunk.energy.specsys = TOPOCENT
Chunk.energy.axis.axis.ctype = WAVE
Chunk.energy.axis.axis.cunit = nm
Chunk.energy.axis.range.start.pix = 0.5
Chunk.energy.axis.range.start.val = _get_energy_chunk_range_start_val(header)
Chunk.energy.axis.range.end.pix = 1.5
Chunk.energy.axis.range.end.val = _get_energy_chunk_range_start_val(header)
extension 9:
Chunk.energy.specsys = TOPOCENT
Chunk.energy.axis.axis.ctype = WAVE
Chunk.energy.axis.axis.cunit = nm
Chunk.energy.axis.range.start.pix = 0.5
Chunk.energy.axis.range.start.val = _get_energy_chunk_range_start_val(header)
Chunk.energy.axis.range.end.pix = 1.5
Chunk.energy.axis.range.end.val = _get_energy_chunk_range_start_val(header)
extension 10:
Chunk.energy.specsys = TOPOCENT
Chunk.energy.axis.axis.ctype = WAVE
Chunk.energy.axis.axis.cunit = nm
Chunk.energy.axis.range.start.pix = 0.5
Chunk.energy.axis.range.start.val = _get_energy_chunk_range_start_val(header)
Chunk.energy.axis.range.end.pix = 1.5
Chunk.energy.axis.range.end.val = _get_energy_chunk_range_start_val(header)
extension 11:
Chunk.energy.specsys = TOPOCENT
Chunk.energy.axis.axis.ctype = WAVE
Chunk.energy.axis.axis.cunit = nm
Chunk.energy.axis.range.start.pix = 0.5
Chunk.energy.axis.range.start.val = _get_energy_chunk_range_start_val(header)
Chunk.energy.axis.range.end.pix = 1.5
Chunk.energy.axis.range.end.val = _get_energy_chunk_range_start_val(header)
extension 12:
Chunk.time.exposure = get_exposure(uri)
Chunk.time.resolution = get_exposure(uri)
Chunk.time.axis.axis.ctype = TIME
Chunk.time.axis.axis.cunit = d
Chunk.time.axis.error.syser = 1e-07
Chunk.time.axis.error.rnder = 1e-07
Chunk.time.axis.function.naxis = 1
Chunk.time.axis.function.delta = get_time_delta(header)
Chunk.time.axis.function.refCoord.pix = 0.5
Chunk.time.axis.function.refCoord.val = get_time_function_val(header)
