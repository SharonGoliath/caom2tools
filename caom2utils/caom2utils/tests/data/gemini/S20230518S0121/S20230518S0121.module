from caom2utils.blueprints import _to_float


def get_exposure(uri):
    return 1200.0


# def _get_energy_chunk_resolving_power(header):
#     start = 76000.0
#     resolut = self._headers[0].get('RESOLUT')
#     if resolut == 'Standard':
#         start = 56000.0

#     red_ccds = self._headers[0].get('REDCCDS')
#     blue_ccds = self._headers[0].get('BLUCCDS')

#     factor_1 = _to_float(red_ccds.split('x')[0])
#     factor_2 = _to_float(blue_ccds.split('x')[0])
#     if factor_1 == factor_2:
#         factor = factor_1
#     else:
#         raise Exception(f'Factors do not match. Blue {factor_2}, Red {factor_1}')
#     return start / factor


def _get_energy_chunk_range_end_val(header):
    result = None
    camera = header.get('CAMERA')
    if camera == 'RED':
        result = 1060.0  # nm
    elif camera == 'BLUE':
        result = 530.0  # nm
    return result


def _get_energy_chunk_range_start_val(header):
    result = None
    camera = header.get('CAMERA')
    if camera == 'RED':
        result = 520.0  # nm
    elif camera == 'BLUE':
        result = 347.0  # nm
    return result
