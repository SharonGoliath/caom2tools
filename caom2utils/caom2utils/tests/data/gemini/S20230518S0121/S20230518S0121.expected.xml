<?xml version='1.0' encoding='UTF-8'?>
<caom2:Observation xmlns:caom2="http://www.opencadc.org/caom2/xml/v2.4" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="caom2:SimpleObservation" caom2:id="65e6d9b6-ad7c-413d-b718-aadfd1c3058d" caom2:metaProducer="GEMINI2caom2/0.0.0">
  <caom2:collection>GEMINI</caom2:collection>
  <caom2:observationID>GS-2023A-SV-101-13-009</caom2:observationID>
  <caom2:metaRelease>2023-05-18T09:12:54.000</caom2:metaRelease>
  <caom2:algorithm>
    <caom2:name>exposure</caom2:name>
  </caom2:algorithm>
  <caom2:type>OBJECT</caom2:type>
  <caom2:intent>science</caom2:intent>
  <caom2:target>
    <caom2:name>CD-38 245</caom2:name>
  </caom2:target>
  <caom2:telescope>
    <caom2:name>Gemini-South</caom2:name>
    <caom2:geoLocationX>1820184.055747326</caom2:geoLocationX>
    <caom2:geoLocationY>-5208320.066618682</caom2:geoLocationY>
    <caom2:geoLocationZ>-3194829.197152751</caom2:geoLocationZ>
  </caom2:telescope>
  <caom2:instrument>
    <caom2:name>GHOST</caom2:name>
  </caom2:instrument>
  <caom2:planes>
    <caom2:plane caom2:id="16d32c9d-ee38-40c3-9072-e120d33a269c" caom2:metaProducer="GEMINI2caom2/0.0.0">
      <caom2:productID>GS-2023A-SV-101-13-009</caom2:productID>
      <caom2:metaRelease>2023-05-18T09:12:54.000</caom2:metaRelease>
      <caom2:dataRelease>2023-05-18T09:12:54.000</caom2:dataRelease>
      <caom2:dataProductType>image</caom2:dataProductType>
      <caom2:calibrationLevel>1</caom2:calibrationLevel>
      <caom2:provenance>
        <caom2:name>Gemini Observatory Data</caom2:name>
        <caom2:project>Gemini Archive</caom2:project>
        <caom2:producer>Gemini Observatory</caom2:producer>
        <caom2:reference>http://archive.gemini.edu/searchform/GS-2023A-SV-101-13-009</caom2:reference>
      </caom2:provenance>
      <caom2:artifacts>
        <caom2:artifact caom2:id="74c274fe-4a00-4345-bd8e-46bcea3d6dfc" caom2:metaProducer="GEMINI2caom2/0.0.0">
          <caom2:uri>cadc:GEMINI/S20230518S0121.fits</caom2:uri>
          <caom2:productType>science</caom2:productType>
          <caom2:releaseType>data</caom2:releaseType>
          <caom2:contentType>application/fits</caom2:contentType>
          <caom2:contentLength>18383040</caom2:contentLength>
          <caom2:contentChecksum>md5:916912dfa1bc11c82dea4f7171876fe2</caom2:contentChecksum>
          <caom2:parts>
            <caom2:part caom2:id="dece16d3-221a-4865-bb58-7b438c7b761f">
              <caom2:name>0</caom2:name>
              <caom2:chunks/>
            </caom2:part>
            <caom2:part caom2:id="24596ef9-e244-4de4-bed0-93d08397cd6c">
              <caom2:name>1</caom2:name>
              <caom2:chunks>
                <caom2:chunk caom2:id="610bfaa3-a58b-4e78-bce6-0e51b871be2b" caom2:metaProducer="GEMINI2caom2/0.0.0">
                  <caom2:naxis>2</caom2:naxis>
                  <caom2:timeAxis>4</caom2:timeAxis>
                  <caom2:time>
                    <caom2:axis>
                      <caom2:axis>
                        <caom2:ctype>TIME</caom2:ctype>
                        <caom2:cunit>d</caom2:cunit>
                      </caom2:axis>
                      <caom2:error>
                        <caom2:syser>1e-07</caom2:syser>
                        <caom2:rnder>1e-07</caom2:rnder>
                      </caom2:error>
                      <caom2:function>
                        <caom2:naxis>1</caom2:naxis>
                        <caom2:delta>0.00020833333333331</caom2:delta>
                        <caom2:refCoord>
                          <caom2:pix>0.5</caom2:pix>
                          <caom2:val>60082.38395833333</caom2:val>
                        </caom2:refCoord>
                      </caom2:function>
                    </caom2:axis>
                    <caom2:timesys>UTC</caom2:timesys>
                    <caom2:exposure>1200.0</caom2:exposure>
                    <caom2:resolution>1200.0</caom2:resolution>
                  </caom2:time>
                </caom2:chunk>
              </caom2:chunks>
            </caom2:part>
            <caom2:part caom2:id="f38ad61c-973c-4506-97fc-8c05d587160a">
              <caom2:name>2</caom2:name>
              <caom2:chunks/>
            </caom2:part>
            <caom2:part caom2:id="0bed86e3-92ac-4fcd-b99a-39a38d4e90f6">
              <caom2:name>3</caom2:name>
              <caom2:chunks>
                <caom2:chunk caom2:id="56ea28e5-c79c-430f-8fcf-d7ffc87e3765" caom2:metaProducer="GEMINI2caom2/0.0.0">
                  <caom2:naxis>2</caom2:naxis>
                  <caom2:energy>
                    <caom2:axis>
                      <caom2:axis>
                        <caom2:ctype>WAVE</caom2:ctype>
                        <caom2:cunit>nm</caom2:cunit>
                      </caom2:axis>
                      <caom2:range>
                        <caom2:start>
                          <caom2:pix>0.5</caom2:pix>
                          <caom2:val>347.0</caom2:val>
                        </caom2:start>
                        <caom2:end>
                          <caom2:pix>1.5</caom2:pix>
                          <caom2:val>347.0</caom2:val>
                        </caom2:end>
                      </caom2:range>
                    </caom2:axis>
                    <caom2:specsys>TOPOCENT</caom2:specsys>
                  </caom2:energy>
                </caom2:chunk>
              </caom2:chunks>
            </caom2:part>
            <caom2:part caom2:id="6fd23f24-3715-486f-afe0-679ff44a6704">
              <caom2:name>4</caom2:name>
              <caom2:chunks>
                <caom2:chunk caom2:id="b41013a1-7b5a-4f14-88fb-c39186a29abd" caom2:metaProducer="GEMINI2caom2/0.0.0">
                  <caom2:naxis>2</caom2:naxis>
                  <caom2:energy>
                    <caom2:axis>
                      <caom2:axis>
                        <caom2:ctype>WAVE</caom2:ctype>
                        <caom2:cunit>nm</caom2:cunit>
                      </caom2:axis>
                      <caom2:range>
                        <caom2:start>
                          <caom2:pix>0.5</caom2:pix>
                          <caom2:val>347.0</caom2:val>
                        </caom2:start>
                        <caom2:end>
                          <caom2:pix>1.5</caom2:pix>
                          <caom2:val>347.0</caom2:val>
                        </caom2:end>
                      </caom2:range>
                    </caom2:axis>
                    <caom2:specsys>TOPOCENT</caom2:specsys>
                  </caom2:energy>
                </caom2:chunk>
              </caom2:chunks>
            </caom2:part>
            <caom2:part caom2:id="c2b633f0-66dd-4ad3-b1d8-af4708648bdc">
              <caom2:name>5</caom2:name>
              <caom2:chunks>
                <caom2:chunk caom2:id="8f8b93e2-4ed0-4d5d-bf71-b263d929fca9" caom2:metaProducer="GEMINI2caom2/0.0.0">
                  <caom2:naxis>2</caom2:naxis>
                  <caom2:energy>
                    <caom2:axis>
                      <caom2:axis>
                        <caom2:ctype>WAVE</caom2:ctype>
                        <caom2:cunit>nm</caom2:cunit>
                      </caom2:axis>
                      <caom2:range>
                        <caom2:start>
                          <caom2:pix>0.5</caom2:pix>
                          <caom2:val>347.0</caom2:val>
                        </caom2:start>
                        <caom2:end>
                          <caom2:pix>1.5</caom2:pix>
                          <caom2:val>347.0</caom2:val>
                        </caom2:end>
                      </caom2:range>
                    </caom2:axis>
                    <caom2:specsys>TOPOCENT</caom2:specsys>
                  </caom2:energy>
                </caom2:chunk>
              </caom2:chunks>
            </caom2:part>
            <caom2:part caom2:id="9f26fcb0-19d0-417b-a9aa-a04e5c3a28f7">
              <caom2:name>6</caom2:name>
              <caom2:chunks>
                <caom2:chunk caom2:id="4b520119-2995-42f6-8f83-bf60bda30b94" caom2:metaProducer="GEMINI2caom2/0.0.0">
                  <caom2:naxis>2</caom2:naxis>
                  <caom2:energy>
                    <caom2:axis>
                      <caom2:axis>
                        <caom2:ctype>WAVE</caom2:ctype>
                        <caom2:cunit>nm</caom2:cunit>
                      </caom2:axis>
                      <caom2:range>
                        <caom2:start>
                          <caom2:pix>0.5</caom2:pix>
                          <caom2:val>347.0</caom2:val>
                        </caom2:start>
                        <caom2:end>
                          <caom2:pix>1.5</caom2:pix>
                          <caom2:val>347.0</caom2:val>
                        </caom2:end>
                      </caom2:range>
                    </caom2:axis>
                    <caom2:specsys>TOPOCENT</caom2:specsys>
                  </caom2:energy>
                </caom2:chunk>
              </caom2:chunks>
            </caom2:part>
            <caom2:part caom2:id="ff7df128-7f02-45a1-a584-b353e10135b6">
              <caom2:name>7</caom2:name>
              <caom2:chunks/>
            </caom2:part>
            <caom2:part caom2:id="f63c011b-251e-4357-bb36-23a3c5878f87">
              <caom2:name>8</caom2:name>
              <caom2:chunks>
                <caom2:chunk caom2:id="e233dcb3-ed87-4f6f-9653-2882905f6561" caom2:metaProducer="GEMINI2caom2/0.0.0">
                  <caom2:naxis>2</caom2:naxis>
                  <caom2:energy>
                    <caom2:axis>
                      <caom2:axis>
                        <caom2:ctype>WAVE</caom2:ctype>
                        <caom2:cunit>nm</caom2:cunit>
                      </caom2:axis>
                      <caom2:range>
                        <caom2:start>
                          <caom2:pix>0.5</caom2:pix>
                          <caom2:val>520.0</caom2:val>
                        </caom2:start>
                        <caom2:end>
                          <caom2:pix>1.5</caom2:pix>
                          <caom2:val>520.0</caom2:val>
                        </caom2:end>
                      </caom2:range>
                    </caom2:axis>
                    <caom2:specsys>TOPOCENT</caom2:specsys>
                  </caom2:energy>
                </caom2:chunk>
              </caom2:chunks>
            </caom2:part>
            <caom2:part caom2:id="f6a462f8-4b5f-4d77-8178-b998face8af1">
              <caom2:name>9</caom2:name>
              <caom2:chunks>
                <caom2:chunk caom2:id="7baad380-cce1-4566-adb8-43a18f1078a6" caom2:metaProducer="GEMINI2caom2/0.0.0">
                  <caom2:naxis>2</caom2:naxis>
                  <caom2:energy>
                    <caom2:axis>
                      <caom2:axis>
                        <caom2:ctype>WAVE</caom2:ctype>
                        <caom2:cunit>nm</caom2:cunit>
                      </caom2:axis>
                      <caom2:range>
                        <caom2:start>
                          <caom2:pix>0.5</caom2:pix>
                          <caom2:val>520.0</caom2:val>
                        </caom2:start>
                        <caom2:end>
                          <caom2:pix>1.5</caom2:pix>
                          <caom2:val>520.0</caom2:val>
                        </caom2:end>
                      </caom2:range>
                    </caom2:axis>
                    <caom2:specsys>TOPOCENT</caom2:specsys>
                  </caom2:energy>
                </caom2:chunk>
              </caom2:chunks>
            </caom2:part>
            <caom2:part caom2:id="5c1e0386-106f-4fc4-bad6-4f77a178d7f2">
              <caom2:name>10</caom2:name>
              <caom2:chunks>
                <caom2:chunk caom2:id="e12f77cf-8660-4aec-b00c-2650092573bc" caom2:metaProducer="GEMINI2caom2/0.0.0">
                  <caom2:naxis>2</caom2:naxis>
                  <caom2:energy>
                    <caom2:axis>
                      <caom2:axis>
                        <caom2:ctype>WAVE</caom2:ctype>
                        <caom2:cunit>nm</caom2:cunit>
                      </caom2:axis>
                      <caom2:range>
                        <caom2:start>
                          <caom2:pix>0.5</caom2:pix>
                          <caom2:val>520.0</caom2:val>
                        </caom2:start>
                        <caom2:end>
                          <caom2:pix>1.5</caom2:pix>
                          <caom2:val>520.0</caom2:val>
                        </caom2:end>
                      </caom2:range>
                    </caom2:axis>
                    <caom2:specsys>TOPOCENT</caom2:specsys>
                  </caom2:energy>
                </caom2:chunk>
              </caom2:chunks>
            </caom2:part>
            <caom2:part caom2:id="7d2e132c-7b18-4c58-9e13-b1fd87a3d216">
              <caom2:name>11</caom2:name>
              <caom2:chunks>
                <caom2:chunk caom2:id="8d8d190d-71dd-40fa-bdb8-575842ad5e4a" caom2:metaProducer="GEMINI2caom2/0.0.0">
                  <caom2:naxis>2</caom2:naxis>
                  <caom2:energy>
                    <caom2:axis>
                      <caom2:axis>
                        <caom2:ctype>WAVE</caom2:ctype>
                        <caom2:cunit>nm</caom2:cunit>
                      </caom2:axis>
                      <caom2:range>
                        <caom2:start>
                          <caom2:pix>0.5</caom2:pix>
                          <caom2:val>520.0</caom2:val>
                        </caom2:start>
                        <caom2:end>
                          <caom2:pix>1.5</caom2:pix>
                          <caom2:val>520.0</caom2:val>
                        </caom2:end>
                      </caom2:range>
                    </caom2:axis>
                    <caom2:specsys>TOPOCENT</caom2:specsys>
                  </caom2:energy>
                </caom2:chunk>
              </caom2:chunks>
            </caom2:part>
            <caom2:part caom2:id="7cac9fe6-aa7c-47ee-98bf-3136e9f9b5f9">
              <caom2:name>12</caom2:name>
              <caom2:chunks>
                <caom2:chunk caom2:id="80b631cd-670a-459b-bc94-4da560f5cbcf" caom2:metaProducer="GEMINI2caom2/0.0.0">
                  <caom2:naxis>2</caom2:naxis>
                  <caom2:timeAxis>4</caom2:timeAxis>
                  <caom2:time>
                    <caom2:axis>
                      <caom2:axis>
                        <caom2:ctype>TIME</caom2:ctype>
                        <caom2:cunit>d</caom2:cunit>
                      </caom2:axis>
                      <caom2:error>
                        <caom2:syser>1e-07</caom2:syser>
                        <caom2:rnder>1e-07</caom2:rnder>
                      </caom2:error>
                      <caom2:function>
                        <caom2:naxis>1</caom2:naxis>
                        <caom2:delta>0.000208333333333365</caom2:delta>
                        <caom2:refCoord>
                          <caom2:pix>0.5</caom2:pix>
                          <caom2:val>60082.3978587963</caom2:val>
                        </caom2:refCoord>
                      </caom2:function>
                    </caom2:axis>
                    <caom2:timesys>UTC</caom2:timesys>
                    <caom2:exposure>1200.0</caom2:exposure>
                    <caom2:resolution>1200.0</caom2:resolution>
                  </caom2:time>
                </caom2:chunk>
              </caom2:chunks>
            </caom2:part>
          </caom2:parts>
        </caom2:artifact>
      </caom2:artifacts>
    </caom2:plane>
  </caom2:planes>
</caom2:Observation>
